src/GNDStk/BodyText.hpp

```diff
@@ -57,56 +45,74 @@ class BodyText {
+   // We'll allow callers to set this by using a setter.
+   std::string rawstring;
+
+   // Vector of <several possibilities>.
+   // Mutable, so that we can defer processing of the raw string into
+   // a vector until, and unless, a caller *asks* for the vector.
+   mutable VariantOfVectors variant;
 
    // Parameters that affect interpretation of the raw string:
-   // vars { length, start, and valueType }
+   //    struct vars { length, start, valueType }
+   // Includes public getters and setters for those.
    #include "GNDStk/BodyText/src/params.hpp"
 
 public:
 
-   enum class Active {
-      string,
-      vector
-   };
-   mutable Active active = Active::string;
-
    // trim
-   // This indicates whether the process of converting a BodyText object back
-   // into data - in a Node - should, or should not, trim zeros from the start
-   // and/or the end of the vector.
+   // Flag: should the conversion of BodyText data back into textual data, in
+   // a Node, trim zeros from the start and end of the output?
    mutable bool trim = true;
 
-   // Getters and setters for length, start, and valueType:
-   #include "GNDStk/BodyText/src/setters.hpp"
-
    // Getters and setters for the raw string:
    #include "GNDStk/BodyText/src/string.hpp"
 
-   // Simple member functions:
-   // clear() and size()
-   #include "GNDStk/BodyText/src/functions.hpp"
-
-   // Various get() functions, including type-specific ones (doubles() etc.)
+   // clear()
+   // Clears the active vector alternative in the variant.
+   BodyText &clear()
+   {
+      std::visit([](auto &&alt) { alt.clear(); }, variant);
+      return *this;
+   }
+
+   // size()
+   // Returns the size of the active vector alternative in the variant.
+   // Depending on what actions someone has or hasn't performed on the current
+   // BodyText object, size() might or might not reflect the values of length
+   // and/or start, or reflect the current contents of the raw string.
+   std::size_t size() const
+   {
+      return std::visit([](auto &&alt) { return alt.size(); }, variant);
+   }
+
+   // Various vector get() functions, and the type-specific doubles() etc.
    #include "GNDStk/BodyText/src/get.hpp"
 
-   // Read/write data from/to a Node
+   // Read/write data, from/to a Node
    #include "GNDStk/BodyText/src/fromNode.hpp"
    #include "GNDStk/BodyText/src/toNode.hpp"
 
    // Write to ostream
    // Not to be confused with the process of writing data to a Node
    #include "GNDStk/BodyText/src/write.hpp"
 
-   // Get length, start, and valueType from derived-class content,
-   // or possibly put them there (if we determined them from a vector)
+   // Pull/push length/start/valueType from/to derived-class struct content
    #include "GNDStk/BodyText/src/sync.hpp"
 
    // Assignment
-   // From vector or string; the latter == calling our raw string setter
+   // From string or vector; the former == calling our raw string setter
    #include "GNDStk/BodyText/src/assign.hpp"
```

Comment:
Having a BodyText constructor for either vector<T> and string similar to the assignment operator would be appreciated. It'd simplify some of the custom constructors.

Reply:
BodyText actually has no explicitly defined constructors. And, Component (which of course derives from BodyText, and *from* which our Standard Interface classes are derived) has only one constructor - one that must receive the derived-class fields that it links with the multi-query. The way things are set up with these classes, I think we're talking here about `finish()` functions, not so much constructors proper. (Issues involving order of initialization during construction have to be considered here, and are the root cause behind the need for the `finish()` functions in the first place.) I do have a `finish(vector)` function available now. Maybe there's a need for a `finish(string)`. Let me think about this.

Reply:
One thing I think we agreed on, in a discussion a little while back, is that we'll work for now on the assumption that length and start are parameters for saving space (by allowing for zeros to be removed) in GNDS files. But, when we're storing and working with a vector, we'll say that all values (including leading and trailing 0s) are actually in the vector, in memory. This saves us from dealing with extra complexities in the BodyText class, and from ambiguities such as what, say, vector element `n` would mean if `start` is nonzero. Would a user want the actual thing in element `[n]` of the vector, if they asked for it, or would they really want `[start+n]`? Anyway, the way the class works, the vector has any necessary zeros, and start should be considered to be 0. The code will automatically deal with leading and trailing zeros when dealing with the string form.

This probably has ramifications for what's in the Values `custom.hpp` at the moment. For the construction from vector, we should say it's a vector with all elements, and dispense with `length`, `start`, and `valueType`.